src/test/java/htsjdk/samtools/CRAMIndexQueryTest.java

```diff
@@ -569,46 +607,95 @@ public void testQueryUnmapped(
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"),
+                        "Mito", 631, 24},
+                {new File(TEST_DATA_DIR, "mitoAlignmentStartTestGATKGen.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"),
+                        "Mito", 631, 24},
+        };
+    }
+
+    @Test(dataProvider="alignmentStartQueries")
+    public void testQueryAlignmentStart(
+            final File cramFileName,
+            final File referenceFileName,
+            final String queryContig,
+            final int alignmentStart,
+            final int expectedReadCount) throws IOException
+    {
+        SamReaderFactory factory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.LENIENT);
+        if (referenceFileName != null) {
+            factory = factory.referenceSequence(referenceFileName);
+        }
+        try (final SamReader reader = factory.open(cramFileName);
+             final CloseableIterator<SAMRecord> it = reader.queryAlignmentStart(queryContig, alignmentStart)) {
+            int count = 0;
+            while (it.hasNext()) {
+                it.next();
+                count++;
+            }
+            Assert.assertEquals(count, expectedReadCount);
+        }
+    }
+
     @DataProvider(name = "mateQueries")
     public Object[][] mateQueries() {
         return new Object[][] {
-                {cramQueryWithCRAI, cramQueryReference, "f"},
-                {cramQueryWithLocalCRAI, cramQueryReference, "f"},
-                {cramQueryWithBAI, cramQueryReference, "f"}
+                // cram, reference, query contig, query start, name of read mates at alignment start
+                {cramQueryWithCRAI, cramQueryReference, "20", 100013, "f"},
+                {cramQueryWithLocalCRAI, cramQueryReference, "20", 100013, "f"},
+                {cramQueryWithBAI, cramQueryReference, "20", 100013, "f"},
+                {new File(TEST_DATA_DIR, "mitoAlignmentStartTest.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"),
+                        "Mito", 631, "IL29_4505:7:30:11521:4492#2"},
+                {new File(TEST_DATA_DIR, "mitoAlignmentStartTestGATKGen.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"),
+                        "Mito", 631, "IL29_4505:7:30:11521:4492#2"},
         };
     }
 
     @Test(dataProvider="mateQueries")
     public void testQueryMate(
         final File cramFileName,
         final File referenceFileName,
-        final String expectedName) throws IOException
+        final String queryContig,
+        final int alignmentStart,
+        final String expectedReadName) throws IOException
     {
-        SamReaderFactory factory = SamReaderFactory.makeDefault();
+        SamReaderFactory factory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.LENIENT);
```

Comment:
why lenient?

Reply:
The test file ("mitoAlignmentStartTest.cram"), which was provided by a user who reported originally reported the queryMate bug, contains reads that have a mateReferenceName = "*", but with  mateAlignmentStart != 0; so we have to use ValidationStringency.LENIENT to suppress validation from throwing when it sees these.